core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java

```diff
@@ -167,6 +173,82 @@ public void reportNetworkCheckUsages(Long stackId, List<NodeStatusProto.NetworkD
         }
     }
 
+    public void nodeStatusMeteringReport(Long stackId) {
+        try {
+            Stack stack = stackService.getById(stackId);
+            if (stack.isDatalake()) {
+                LOGGER.debug("Skip metering check as billing is not used for datalake");
+                return;
+            }
+            if (!meteringConfiguration.isEnabled()) {
+                LOGGER.debug("Metering feature is not enabled (or telemetry is empty). Skip metering check.");
+                return;
+            }
+            RPCResponse<NodeStatusProto.NodeStatusReport> rpcResponse = nodeStatusService.getMeteringReport(stackId);
+            if (rpcResponse != null) {
+                LOGGER.debug("Diagnostics metering report response: {}", rpcResponse.getFirstTextMessage());
+                NodeStatusProto.NodeStatusReport nodeStatusReport = rpcResponse.getResult();
+                if (nodeStatusReport != null && CollectionUtils.isNotEmpty(nodeStatusReport.getNodesList())) {
+                    List<NodeStatusProto.NodeStatus> nodeStatuses = nodeStatusReport.getNodesList();
+                    Set<String> errorSet = new HashSet<>();
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getHeartbeatAgentRunning,
+                            "Heartbeat agents running", errorSet);
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getHeartbeatConfig,
+                            "Heartbeat agents configured", errorSet);
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getLoggingServiceRunning,
+                            "Logging agents running", errorSet);
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getLoggingAgentConfig,
+                            "Logging agents configured", errorSet);
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getDatabusReachable,
+                            "DataBus reachable", errorSet);
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getDatabusTestResponse,
+                            "DataBus test response", errorSet);
```

Comment:
this seems a bit C style, would be more object oriented if you would have like an interface/abstract class with multiple implementation all of which would be called and could be injected by spring framework